### ISSUE TITLE
Reports of Windows users losing peers #13431 

### ISSUE THREAD

**Issue by defeedme** [prysmaticlabs/prysm#13936]
### Describe the bug

I spoke too soon. Still having the same problem and even changed ISP and still not helping. I've worked with nishant on this for a long time and we tried almost every trick in the book. At one point after changing some settings on net-time (time drift) it worked perfect for 2 days, which was the longest.. The only thing that halfway works now is to literally shut down the node every 4 hours and re-start using task scheduler.. not a very healthy way to run a validator..

**Background**
Opening this as it's nuanced and haven't been wide spread however it does randomly happen to some windows users
where the OS decides to drop all packets from prysm for some reason. This results in a reduced peer count - and eventually all peers lost. Root cause is currently unknown but opening this as a tracking issue. 

Currently there have been a non trivial amount of users complaining on Windows about their number of peers
slowly reducing to zero. v1.05 has had some important changes with regards to stream management and subnet search.
However none of these components should have had any material impact on discovery. Windows machines being more
susceptible to clock drifts suddenly stop being able to find new peers for some reason, being minutes ahead/behind with respect to the network time should by right not have any effect in finding new peers.

related https://github.com/prysmaticlabs/documentation/issues/891
related https://github.com/prysmaticlabs/prysm/issues/8144

### Has this worked before in a previous version?

```text
It worked for almost 3 years with no issues. Then in december 2023 out of the blue this issue started. Geth would not sync anymore so I switched to nethermind which has been great.
```

### 🔬 Minimal Reproduction

1. start the prysm node

### Error

```text
all peers lost between 1 hour and 1 day.
```

### Platform(s)

Windows 10 (x86) 

### What version of Prysm are you running? (Which release)

latest version

### Anything else relevant (validator index / public key)?

https://beaconcha.in/validator/26606#attestations
![almostperfectuntildec23](https://github.com/prysmaticlabs/prysm/assets/31375635/9ac15c82-2bd5-474f-ba90-1c79123ada81)

**Comment by web3dev2023** [comment]
I am having this same problem of losing peers on prysm and geth for months. I have a Windows 10 and a Windows 11 PC and they were running prysm and geth smoothly before January, until they began losing peers.

This bug does not seem to happen on every Windows machine. I have a Windows server in NYC with the same configs and even the same database files and everything works fine. 

For those who are suffering from this issue, I have a temporary workaround before this bug is fixed(by using lighthouse and static peers):
https://github.com/prysmaticlabs/prysm/issues/13431#issuecomment-2098280439

**Comment by defeedme** [comment]
> I am having this same problem of losing peers on prysm and geth for months. I have a Windows 10 and a Windows 11 PC and they were running prysm and geth smoothly before January, until they began losing peers.
> 
> This bug does not seem to happen on every Windows machine. I have a Windows server in NYC with the same configs and even the same database files and everything works fine.
> 
> For those who are suffering from this issue, I have a temporary workaround before this bug is fixed(by using lighthouse and static peers): [#13431 (comment)](https://github.com/prysmaticlabs/prysm/issues/13431#issuecomment-2098280439)

thanks so much for the workaround! i've been suffering for months lol.. at least now I know I'm not crazy and this is definitely a real issue.. lighthouse here we come

**Comment by anderspatriksvensson** [comment]
Having the EXACT same issue.  Since tesnet and genesis validating without any issues on windows.  Now all of a sudden last few days peers drop to zero and must restart prysm validator to fix the issue.  Even with a restart, it fails within a day or so and then must restart again.  Tagging myself in here to see if a real solution comes along, not a big fan of setting static nodes.

**Comment by hzysvilla** [comment]
#14025 a new report to the same issue.

**Comment by prestonvanloon** [comment]
@nisdas is this fixed with https://github.com/prysmaticlabs/prysm/pull/14052 ?

**Comment by nisdas** [comment]
This is fixed by #14052, since that has been merged in I will be closing the issue now.
